src/components/Clock.tsx

```diff
@@ -0,0 +1,54 @@
+import { Component } from 'react';
+
+type State = {
+  date: Date,
+};
+
+type Props = {
+  name: string,
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ date: new Date() });
+    }, 1000);
+  }
+
+  componentDidUpdate(_: never, prevState: State) {
+    const { date } = this.state;
+
+    if (date !== prevState.date) {
+      window.console.info(date.toUTCString().slice(-12, -4));
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+  }
+
+  render() {
+    const { date } = this.state;
+    const { name } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {name}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {date.toUTCString().slice(-12, -4)}
```

Comment:
You should follow DRY principle here. That means Don't Repeat Yourself. You already do it in your code, so why won't we create a separate method for that?